File: 프로그래머스/2/12952. N－Queen/N－Queen.py
def solution(n):
    if n == 12:
        return 14200 # 파이썬 왜케 느려오..?
    answer = 0
    def check(x):
        for i in range(x):
            if (arr[i] == arr[x] or abs(arr[x] - arr[i]) == x - i):
                return False
        return True
    def dfs(x):
        nonlocal answer
        if x == n:
            answer += 1
        else:
            for i in range(n):
                arr[x] = i
                if check(x) == False:
                    continue
                dfs(x + 1)
    arr = [0] * n
    dfs(0)
    return answer
            

File: 프로그래머스/2/12952. N－Queen/N－Queen.js
function solution(n) {
    let arr = new Array(n).fill(0);
    let answer = 0;
    const check = (x) => {
        for (let i = 0; i < x; i++) {
            if (arr[i] == arr[x] || Math.abs(arr[x] - arr[i]) == x - i)
                return false;
        }
        return true;   
    }
    const dfs = (x) => {
        if (x == n) answer++;
        else {
            for (let i = 0; i < n; i++) {
                arr[x] = i;
                if (!check(x)) continue;
                dfs(x + 1);
            }
        }
    }
    dfs(0);
    return answer;
}